mods/boats/init.lua

```diff
@@ -181,15 +181,12 @@ function boat.on_step(self, dtime)
 		return
 	end
 	local s = get_sign(self.v)
-	self.v = self.v - dtime * 0.6 * s
+	self.v = self.v - dtime * 0.2 * self.v * self.v
 	if s ~= get_sign(self.v) then
 		self.object:set_velocity({x = 0, y = 0, z = 0})
 		self.v = 0
 		return
 	end
-	if math.abs(self.v) > 5 then
-		self.v = 5 * get_sign(self.v)
-	end
```

Comment:
We don't need maximum value anymore, as Drag Force is dependent from Speed^2.
It means that it the boat speed is already limited.